src/test/java/com/cloudbees/plugins/credentials/impl/CertificateCredentialsImplFIPSTest.java

```diff
@@ -0,0 +1,80 @@
+package com.cloudbees.plugins.credentials.impl;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.text.StringEscapeUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.RealJenkinsRule;
+
+import hudson.ExtensionList;
+import hudson.model.Descriptor;
+import hudson.util.FormValidation;
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+
+public class CertificateCredentialsImplFIPSTest {
+
+    @Rule
+    public RealJenkinsRule rule = new RealJenkinsRule().javaOptions("-Djenkins.security.FIPS140.COMPLIANCE=true");
```

Comment:
please use [`RealJenkinsRile.withFipsEnabled`](https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/RealJenkinsRule.html#withFIPSEnabled()) and always specify some memory options so this does not use default ergonomics and grab a huge portion of available memory.